### ISSUE TITLE
Introduce an asynchronous, streaming runCore

### ISSUE THREAD

**Issue by fpapado** [fpapado/accessible-pipeline#13]
The current runCore function buffers all the results in memory, until everything is done.
Only at the end, does it actually give the results to the caller.

This is fine for small workloads, but it can be prohibitive for larger ones, especially given [the large size of AxE results](#4).

Another issue with buffering is that streaming workloads (i.e. the CLI "reporter" API) need some hard-coded support for reporting progress. You can see this in the current implementation as a custom pino logger. While I like the current setup (unix-style piping is cool!), I think we could unify this use-case with a streaming API at the runCore level.

The current runCore API (that buffers) can be built on top of a streaming one, by collecting all of the results, and flushing at the end.

## Potential next steps

Something along these lines could work:
- Rename the current `runCore` to `runCoreStreaming`, that outputs a stream. This could be either a Node stream, or an RxJS Observable. I'm more familiar with Observables, so I think I'll go with that. Instead of appending to results and returning, instead return the Observable, and push to it as we go.
- Add a new `runCore` function, that calls `runCoreStreaming`, and collects the results. Return those results once the Observable is done.
- Finally, consider swapping `runCore` to be `runCoreStreaming`. Rename the return-based `runCore` to `runCoreBuffered`, or so.
- Add docs with this behaviour

## Future work

This mostly concerns the CLI use case:
- Remove the current "streaming API" logging. Move its events to the CLI module, or a module on top of core.
- Consider whether to still use sub-processing in the CLI, or whether a function call would suffice.
